commands/info.go

```diff
@@ -21,47 +21,101 @@ var (
 	}
 )
 
+// infoArguments represents the arguments we can make use of in this command
+type infoArguments struct {
+	token       string
+	verbose     bool
+	apiEndpoint string
+}
+
+// defaultInfoArguments returns an infoArguments object populated by the user's
+// command line arguments
+func defaultInfoArguments() infoArguments {
+	return infoArguments{
+		token:       cmdToken,
+		verbose:     cmdVerbose,
+		apiEndpoint: cmdAPIEndpoint,
+	}
+}
+
+// infoResult is the struct used to return all the info we might want to print
+type infoResult struct {
+	apiEndpoint     string
+	email           string
+	token           string
+	version         string
+	buildDate       string
+	configFilePath  string
+	kubeConfigPaths []string
+}
+
 func init() {
 	RootCommand.AddCommand(InfoCommand)
 }
 
 // printInfo prints some information on the current user and configuration
 func printInfo(cmd *cobra.Command, args []string) {
 
+	infoArgs := defaultInfoArguments()
+	result := info(infoArgs)
+
 	output := []string{}
 
-	if config.Config.Email == "" {
+	output = append(output, color.YellowString("API endpoint:")+"|"+color.CyanString(result.apiEndpoint))
+
+	if result.email == "" {
 		output = append(output, color.YellowString("Email:")+"|"+"n/a")
 	} else {
 		output = append(output, color.YellowString("Email:")+"|"+color.CyanString(config.Config.Email))
 	}
 
-	if cmdVerbose {
-		if cmdToken != "" {
-			output = append(output, color.YellowString("Auth token:")+"|"+color.CyanString(cmdToken))
-		} else if config.Config.Token != "" {
-			output = append(output, color.YellowString("Auth token:")+"|"+color.CyanString(config.Config.Token))
+	if infoArgs.verbose {
+		if result.token != "" {
+			output = append(output, color.YellowString("Auth token:")+"|"+color.CyanString(result.token))
 		} else {
 			output = append(output, color.YellowString("Auth token:")+"|n/a")
 		}
 	}
 
-	output = append(output, color.YellowString("%s version:", config.ProgramName)+"|"+color.CyanString(config.Version))
-	output = append(output, color.YellowString("%s build:", config.ProgramName)+"|"+color.CyanString(config.BuildDate))
+	output = append(output, color.YellowString("%s version:", config.ProgramName)+"|"+color.CyanString(result.version))
+	output = append(output, color.YellowString("%s build:", config.ProgramName)+"|"+color.CyanString(result.buildDate))
+	output = append(output, color.YellowString("Config path:")+"|"+color.CyanString(result.configFilePath))
+
+	// kubectl configuration paths
+	output = append(output, color.YellowString("kubectl config path:")+"|"+color.CyanString(strings.Join(result.kubeConfigPaths, ", ")))
+
+	fmt.Println(columnize.SimpleFormat(output))
+}
+
+// info gets all the information we'd like to show with the "info" command
+// and returns it as a struct
+func info(args infoArguments) infoResult {
+	result := infoResult{}
+
+	result.apiEndpoint = config.DefaultAPIEndpoint
```

Comment:
Not sure if mixing global and parameter variables is a good idea in this case. The signature looks like pure function.